app/apps/server/communication/rest.js

```diff
@@ -392,6 +398,32 @@ export class AppsRestApi {
 			},
 		});
 
+		this.api.addRoute(':id/sync', { authRequired: true, permissionsRequired: ['manage-apps'] }, {
+			post() {
+				const baseUrl = orchestrator.getMarketplaceUrl();
+
+				const headers = getDefaultHeaders();
+				const token = getWorkspaceAccessToken();
+				if (token) {
+					headers.Authorization = `Bearer ${ token }`;
+				}
+
+				const [workspaceIdSetting] = Settings.findById('Cloud_Workspace_Id').fetch();
+
+				const result = HTTP.get(`${ baseUrl }/v1/workspaces/${ workspaceIdSetting.value }/apps/${ this.urlParams.id }`, {
+					headers,
+				});
+
+				if (result.statusCode !== 200) {
```

Comment:
Need to add debug logging here because the errors are getting lost when the Marketplace API doesn't return a success. And anywhere else in this file that the Marketplace is called and we handle the "non success" results.